File: admin/frontend/src/api/nodes.ts
import { apiFetch } from "./client";
import type { NodeStatusMap, LogEntry } from "../types";

export function getNodes(): Promise<NodeStatusMap> {
  return apiFetch<NodeStatusMap>("/api/nodes");
}

export function getLogs(node?: string, limit = 100): Promise<LogEntry[]> {
  const params = new URLSearchParams();
  if (node) params.set("node", node);
  params.set("limit", String(limit));
  return apiFetch<LogEntry[]>(`/api/logs?${params.toString()}`);
}

File: admin/app/routers/nodes.py
from fastapi import APIRouter, Request

router = APIRouter()


@router.get("/api/nodes")
def get_nodes(request: Request):
    return request.app.state.tracker.get_nodes()


@router.get("/api/logs")
def get_logs(request: Request, node: str | None = None, limit: int = 100):
    return request.app.state.tracker.get_recent_logs(node=node, limit=limit)
